client/src/js/services/PrototypeApiService.js

```diff
@@ -244,17 +244,32 @@ function PrototypeApiService($http, util) {
 
     // append the id to the base url
     var target = this.url.concat('reports/', param);
+    return reportBuilder(this, param, filetype);
+  }
+
+  /**
+   * Receipt
+   */
+  function receipt(param, filetype) {
+
+    // append the id to the base url
+    var target = this.url.concat('receipts/', param);
+    return reportBuilder(this, param, filetype);
+  }
+
+  /** report builder */
+  function reportBuilder(container, param, filetype) {
 
     // filetype setup
     var responseType = filetype === 'pdf' ? 'arraybuffer' : null;
     var params = { renderer: filetype };
 
     // send the GET request
-    return this.$http.get(target, {
+    return container.$http.get(target, {
```

Comment:
You don't need `container`.  Just use `$http`, it is available in the current scope.